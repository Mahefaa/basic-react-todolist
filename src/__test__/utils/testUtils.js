export const TASK_TODO_STATUS = "TODO";
export const TASK_DONE_STATUS = "DONE";