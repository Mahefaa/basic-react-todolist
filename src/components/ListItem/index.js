import style from './index.module.css'

export default function ListItem({task, index, handleCheck}) {
    return (
        <div key={task.id}>
            {task.status === "TODO" &&
                <input type={"checkbox"} value={index} onClick={handleCheck} data-testid={`TODO-${index}`}/>}
            <span className={`${style.task__name} ${task.status === "DONE" ? style.done : style.todo}`}>
                {task.name}
            </span>
        </div>
    );
}