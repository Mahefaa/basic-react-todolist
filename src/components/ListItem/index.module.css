.done {
    color: #2e913d;
}

.todo {
    color: #d42527;
}

.task__name {
    text-effect: engrave;
    font-weight: bolder;
}