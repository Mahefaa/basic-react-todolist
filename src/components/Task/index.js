export default class Task {
    id;
    name;
    status;

    constructor(name, status = "TODO") {
        this.id = Date.now();
        this.name = name;
        this.status = status;
    }
}