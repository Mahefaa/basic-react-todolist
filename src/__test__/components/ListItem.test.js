import renderer from 'react-test-renderer';
import Task from "../../components/Task";
import ListItem from "../../components/ListItem";
import {cleanup} from "@testing-library/react";
import {TASK_DONE_STATUS, TASK_TODO_STATUS} from "../utils/testUtils";

afterEach(cleanup);
describe("ListItem component is rendered correctly", () => {
    it("should render a task with status todo", () => {
        const task = {id: 1, name: "name", status: TASK_TODO_STATUS};
        const tree = renderer
            .create(<ListItem task={task} handleCheck={null} index={null}/>)
            .toJSON();

        expect(tree).toMatchSnapshot();
    })
    it("should render a task with status done", () => {
        const task = new Task("name", TASK_DONE_STATUS);

        const tree = renderer
            .create(<ListItem task={task} handleCheck={null} index={null}/>)
            .toJSON();

        expect(tree).toMatchSnapshot();
    })
})