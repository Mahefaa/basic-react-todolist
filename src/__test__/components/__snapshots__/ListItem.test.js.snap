// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component is rendered correctly should render a task with status done 1`] = `
<div>
  <span
    className="task__name done"
  >
    name
  </span>
</div>
`;

exports[`ListItem component is rendered correctly should render a task with status todo 1`] = `
<div>
  <input
    data-testid="TODO-null"
    onClick={null}
    type="checkbox"
    value={null}
  />
  <span
    className="task__name todo"
  >
    name
  </span>
</div>
`;
